locator/topology.cc

```diff
@@ -495,6 +495,14 @@ std::weak_ordering topology::compare_endpoints(const inet_address& address, cons
     return d1 <=> d2;
 }
 
+void topology::for_each_node(std::function<void(const node*)> func) const {
```

Comment:
Will `nullptr` be ever passed to `func`? If not, better use `const node&`?

Reply:
No, null won't be passed to func.
However, `node*` becomes an commodity by itself and further patches, e.g. converting node operations to use nodes, manage sets of `const node*` as sync_nodes or ignore_nodes, and in these cases we use the node* as-is to see if it's contained in those sets or insert it to sets.  It will be odd to pass a reference just to take it's address in `func`.

I'm contemplating whether it would be a good thing to define locator::node_ptr using const locator::node* so it'd be clearer it is a defined type we use to identify nodes or not.  I think this shows there could be a place for such typedef.

Reply:
Ok. I guess we can thank C++'s weird references for not being able to use them, instead of `node*`.